internal/provider/data_file.go

```diff
@@ -0,0 +1,130 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/neuspaces/terraform-provider-system/internal/client"
+	"github.com/neuspaces/terraform-provider-system/internal/validate"
+	"path"
+)
+
+const dataFileName = "system_file"
+
+func dataFile() *schema.Resource {
+	return &schema.Resource{
+		Description: fmt.Sprintf("`%s` retrieves information about a file on the remote system.", dataFileName),
+
+		ReadContext: dataFileRead,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		SchemaVersion: 1,
+
+		Schema: map[string]*schema.Schema{
+			resourceFileAttrId: {
+				Description: "ID of the file",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			resourceFileAttrPath: {
+				Description: "Absolute path to the file.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			resourceFileAttrMode: {
+				Description:      "Permissions of the file in octal format like `755`. Defaults to the umask of the system.",
+				Type:             schema.TypeString,
+				Computed:         true,
+				ValidateDiagFunc: validate.FileMode(),
```

Comment:
Please remove all `ValidateDiagFunc` in the schema as they are not allowed for a data source. Provider validation tests fail when they are present.